embeddings/model/lightning_module/huggingface_module.py

```diff
@@ -24,11 +25,11 @@ def __init__(
         self.save_hyperparameters({"downstream_model_type": downstream_model_type.__name__})
         self.downstream_model_type = downstream_model_type
         self.config_kwargs = config_kwargs if config_kwargs else {}
+        self.configure_model()
```

Comment:
Why is it moved to `__init__`?

Reply:
`load_from_checkpoint()` method does not work in another case - the model has to be initialized in the `__init__` for this method to work properly and read the saved checkpoint

Reply:
ok but I think's it needs better solution. I would keep this functions call in the `setup` and prepare appropiate configuration for stage  "prediction"? There is special hook that we can use while loading checkpoints  https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.hooks.CheckpointHooks.html#pytorch_lightning.core.hooks.CheckpointHooks.on_load_checkpoint

Reply:
I mean, the initialization of the model in PytorchLightning is usually done in __init__ method if you look at documentation so I don't think there is something wrong with this solution; moving initialization to setup was only done to not ask user to give the num classes arg

Reply:
I dissagree, I consider configuring models etc. as `setup` functionality than `init`

Reply:
Ok but if we using a framework such as `PytorchLightning` we should consider their good practices of designing/writing the code; as I said, if you look at any of the examples in the library the model is almost always initialized in the `__init__` - the setup is used only to build models dynamically or adjust something about them and thus it is not a default place of model building

Reply:
I really don't think we should reinvent their library inside of ours

Reply:
Sorry I didn't mean to sound aggressive. 
1. It was placed in `__init__` at first but then it was moved to `setup()` to not force user to pass the `num_classes` argument. 
2. You are right but then we need to load the whole model with `on_load_checkpoint` method which I would call very unnecessary and undesirable. When moving it to `__init__` we allow for loading from checkpoint easily and stick to the convention.
3. Okay I can change the names to _init_model etc. if that seems as a better solution.